temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -180,149 +243,375 @@ public void handle(LocalActivityTask task) throws Exception {
                   MetricsTag.WORKFLOW_TYPE,
                   activityTask.getWorkflowType().getName()));
 
-      MDC.put(LoggerTag.ACTIVITY_ID, activityTask.getActivityId());
+      int currentAttempt = activityTask.getAttempt();
+
+      MDC.put(LoggerTag.ACTIVITY_ID, activityId);
       MDC.put(LoggerTag.ACTIVITY_TYPE, activityTask.getActivityType().getName());
       MDC.put(LoggerTag.WORKFLOW_ID, activityTask.getWorkflowExecution().getWorkflowId());
       MDC.put(LoggerTag.WORKFLOW_TYPE, activityTask.getWorkflowType().getName());
       MDC.put(LoggerTag.RUN_ID, activityTask.getWorkflowExecution().getRunId());
-
-      ActivityTaskHandler.Result result = null;
       try {
-        result =
-            handleLocalActivity(
-                params.getActivityTask(),
-                System.currentTimeMillis(),
-                params.getLocalRetryThreshold().toMillis(),
-                metricsScope);
+        ScheduledFuture<?> startToCloseTimeoutFuture = null;
+
+        if (activityTask.hasStartToCloseTimeout()) {
+          startToCloseTimeoutFuture =
+              scheduledExecutor.schedule(
+                  new StartToCloseTimeoutHandler(attemptTask),
+                  ProtobufTimeUtils.toJavaDuration(
+                          attemptTask.getAttemptTask().getStartToCloseTimeout())
+                      .toMillis(),
+                  TimeUnit.MILLISECONDS);
+        }
+
+        metricsScope.counter(MetricsType.LOCAL_ACTIVITY_TOTAL_COUNTER).inc(1);
+
+        ActivityTaskHandler.Result activityHandlerResult;
+        Stopwatch sw = metricsScope.timer(MetricsType.LOCAL_ACTIVITY_EXECUTION_LATENCY).start();
+        try {
+          activityHandlerResult =
+              handler.handle(new ActivityTask(activityTask, () -> {}), metricsScope, true);
+        } finally {
+          sw.stop();
+        }
+
+        // Making sure that the result handling code following this statement is mutual exclusive
+        // with the start to close timeout handler.
+        boolean startToCloseTimeoutFired =
+            startToCloseTimeoutFuture != null && !startToCloseTimeoutFuture.cancel(false);
+
```

Comment:
Since this is coming after `handle`, does this mean the start-to-close timeout cannot "interrupt" the running activity if it fires - but does it tell the workflow about the timeout right away?

So here we're just saying "OK we (elsewhere) told the WF the LA timed out, and now we're just discarding the result when the LA eventually finishes"?

Do we try to notify the LA of cancel / interrupt it at all? I'm guessing not since there's not a good mechanism for it